sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/encoders/ProductEncoderSuite.scala

```diff
@@ -215,6 +217,42 @@ class ProductEncoderSuite extends SparkFunSuite {
   protected def encodeDecodeTest[T <: Product : TypeTag](inputData: T) =
```

Comment:
Could we extend encodeDecodeTest to also just try converting `InternalRow => Row` and then `Row => InternalRow` and make sure the results match?  That seems like a good way to get coverage without having to repeat a lot of test code.